src/web/user_agent.rs

```diff
@@ -50,6 +50,12 @@ pub struct DeviceInfo {
     pub os_family: OsFamily,
 }
 
+impl ToString for DeviceInfo {
```

Comment:
`ToString` docs recommend you don't implement it directly, impl `Display` instead.

However I don't think we need it or the new `Location::as_string` at all since this just adds new metrics calls where extras aren't used.

If we did want these extras included in later occurring errors this probably isn't the best place to add them. I don't think they're needed as adM errors like [this one](https://sentry.prod.mozaws.net/operations/contile/issues/11738905/?query=is%3Aunresolved) include the adM url in the error string with that information (we could add that adM url explicitly as an extra but `reqwest::Error` seems to always include it for us anyway)